### ISSUE TITLE
Is 'None' key really valid in YAML for implicit_solvent: section?

### ISSUE THREAD

**Issue by jchodera** [choderalab/yank#530]
I got the following error when I tried to use `None` as a key for `implicit_solvent:`:

```
2016-10-16 20:16:50,068: ERROR - yank.yamlbuild - Solvent vacuum: Wrong keys 'implicit_solvent' in {'nonbonded_method': 'NoCutoff', 'implicit_solvent': None}
ERROR

======================================================================
ERROR: Test the restraints in a protein:ligand system.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/choderaj/github/choderalab/yank/yank.choderalab/Yank/tests/test_restraints.py", line 163, in test_protein_ligand_restraints
    yaml_builder = YamlBuilder(yaml_script % data)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/yank/yamlbuild.py", line 1200, in __init__
    self.parse(yaml_source)
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/yank/yamlbuild.py", line 1261, in parse
    self._db.solvents = self._validate_solvents(yaml_content.get('solvents', {}))
  File "/Users/choderaj/miniconda/lib/python2.7/site-packages/yank/yamlbuild.py", line 1632, in _validate_solvents
    raise YamlParseError('Solvent {}: {}'.format(solvent_id, e.autos[-1]))
YamlParseError: Solvent vacuum: Wrong keys 'implicit_solvent' in {'nonbonded_method': 'NoCutoff', 'implicit_solvent': None}
```

**Comment by jchodera** [comment]
I think omitting `implicit_solvent:` altogether is the solution to get vacuum, but we should [revise our docs on this](http://getyank.org/latest/yamlpages/solvents.html#implicit-solvent) or permit the value of `None` or `null`.

**Comment by andrrizzi** [comment]
I think `implicit_solvent: null` (`None` never works in YAML) is currently **not** supported. If that's in the documentation, it is probably incorrect.

**Comment by jchodera** [comment]
The docs say:
```
Valid Options: [None] / HCT / OBC1 / OBC2 / GBn / GBn2
```

**Comment by Lnaden** [comment]
Note to self to fix this pre 1.0
